ocaml/xapi/monitor_master.ml

```diff
@@ -37,10 +36,19 @@ let update_configuration_from_master () =
     )
 
 let get_pciids vendor device =
+  let id_of_string id =
+    int_of_string_opt ("0x" ^ id)
+  in
+  let (>>=) = Option.bind in
   (* FIXME : put a lazy cache *)
-  let v, d = Pciutil.parse vendor device in
-  (match v with None -> "" | Some x -> x),
-  (match d with None -> "" | Some x -> x)
+  let pci_ids =
+    id_of_string vendor >>= fun vendor_id ->
+    id_of_string device >>= fun device_id ->
+    Pci.(with_access lookup_vendor_name) vendor_id >>= fun vendor_str ->
+    Pci.(with_access lookup_device_name) vendor_id device_id >>= fun device_str ->
+    Some (vendor_str, device_str)
```

Comment:
For names and vendors without a name pcutil returned `Unknown vendor <vendor_id>`, now it returns `Vendor <vendor_id>`.

Unsure if this is bound to break on the upgrade case or even if it's important.

Reply:
Pciutil doesn't depend on libpci (which we know had buggy interfaces with OCaml) so should be more reliable. However we're already using pci elsewhere in xapi, so might as well use it here.